capella2polarion/converters/document_renderer.py

```diff
@@ -0,0 +1,196 @@
+        default_factory=list
+    )
+    heading_ids: list[str] = dataclasses.field(default_factory=list)
+    rendering_layouts: list[polarion_api.RenderingLayout] = dataclasses.field(
+        default_factory=list
+    )
+
+
+class DocumentRenderer(polarion_html_helper.JinjaRendererMixin):
+    """A Renderer class for Polarion documents."""
+
+    def __init__(
+        self,
+        polarion_repository: polarion_repo.PolarionDataRepository,
+        model: capellambse.MelodyModel,
+    ):
+        self.polarion_repository = polarion_repository
+        self.model = model
+        self.jinja_envs: dict[str, jinja2.Environment] = {}
+
+    def setup_env(self, env: jinja2.Environment):
+        """Add globals and filters to the environment."""
+        env.globals["insert_work_item"] = self.__insert_work_item
+        env.globals["heading"] = self.__heading
+        env.filters["link_work_item"] = self.__link_work_item
+
+    def __insert_work_item(
+        self, obj: object, session: RenderingSession, level: int | None = None
+    ) -> str | None:
+        if (obj := self.check_model_element(obj)) is None:
+            return polarion_html_helper.RED_TEXT.format(
+                text="A none model object was passed to insert a work item."
+            )
+
+        if wi := self.polarion_repository.get_work_item_by_capella_uuid(
+            obj.uuid
+        ):
+            layout_index = 0
+            for layout in session.rendering_layouts:
+                if layout.type == wi.type:
+                    break
+                layout_index += 1
+
+            if layout_index >= len(session.rendering_layouts):
+                session.rendering_layouts.append(
+                    polarion_api.RenderingLayout(
+                        type=wi.type,
+                        layouter="section",
+                        label=polarion_html_helper.camel_case_to_words(
+                            wi.type
+                        ),
+                    )
+                )
+
+            custom_info = ""
+            if level is not None:
+                custom_info = f"level={level}|"
+
+            return polarion_html_helper.POLARION_WORK_ITEM_DOCUMENT.format(
+                pid=wi.id, lid=layout_index, custom_info=custom_info
+            )
+
+        return polarion_html_helper.RED_TEXT.format(
+            text=f"Missing WorkItem for UUID {obj.uuid}"
+        )
+
+    def __link_work_item(self, obj: object) -> str | None:
+        if (obj := self.check_model_element(obj)) is None:
+            raise TypeError("object passed was no model element")
+
+        if wi := self.polarion_repository.get_work_item_by_capella_uuid(
+            obj.uuid
+        ):
+            return polarion_html_helper.POLARION_WORK_ITEM_URL.format(
+                pid=wi.id
+            )
+
+        return polarion_html_helper.RED_TEXT.format(
+            text=f"Missing WorkItem for UUID {obj.uuid}"
```

Comment:
```suggestion
            text=f"Missing WorkItem for {obj._short_repr_()!r}"
```

Reply:
No `_short_repr_()` returns a string similar to `<TYPE `NAME` (UUID)>`.

Reply:
Yes and that's the issue. This would be interpreted as as an HTML Element and would be invisible for that reason